File: server/src/entity/Order.ts
import {
  BaseEntity,
  Column,
  CreateDateColumn,
  Entity,
  JoinTable,
  ManyToMany,
  ManyToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';

import {
  IsBoolean,
  IsDate,
  IsEnum,
  IsNumber,
  IsString,
  ValidateNested,
} from 'class-validator';
import { Account } from './Account';
import { MenuItem } from './MenuItem';
import { Shift } from './Shift';
import { Visit } from './Visit';

export enum orderStatus {
  RECEIVE = 'receive',
  PREP = 'prep',
  COOK = 'cook',
  READY = 'ready',
}

@Entity()
export class Order extends BaseEntity {
  @PrimaryGeneratedColumn()
  public id: string;

  @Column({
    type: 'enum',
    enum: orderStatus,
    default: orderStatus.RECEIVE,
  })
  @IsEnum(orderStatus)
  public prepStatus: orderStatus;

  @Column({
    default: 10,
  })
  @IsNumber()
  public orderETA: number;

  @CreateDateColumn()
  @IsDate()
  public start: Date;

  @Column({
    nullable: true,
  })
  @IsDate()
  public end: Date;

  @ManyToOne((type) => Account, (account) => account.assignedOrders)
  @ValidateNested()
  public cookAssigned: Account;

  @Column()
  @IsString()
  public comments: string;

  @ManyToOne((type) => Shift, (shift) => shift.orders)
  public shift: Shift;

  @ManyToOne((type) => Visit, (visit) => visit.orders)
  @JoinTable()
  public visit: Visit;

  @ManyToMany((type) => MenuItem, (menuItem) => menuItem.orders)
  public menuItems: MenuItem[];
}

File: server/python_db_client/order.py
from ientity import IEntity
from jeeves_api_config import ApiRoutes
from enum import Enum
import datetime


class Status(Enum):
    RECEIVE = 'receive'
    PREP = 'prep'
    COOK = 'cook'
    READY = 'ready'


class Order(IEntity):
    id = 0

    def __init__(self,
                 status=Status.RECEIVE,
                 eta=10,
                 cook=1,
                 comments='Terrible restaurant.',
                 shift=1,
                 visit=1,
                 start_time=datetime.datetime.now().isoformat(),
                 end_time=datetime.datetime.now().isoformat(),
                 id=None):
        super().__init__(True, id)
        self._json['prepStatus'] = status.value
        self._json['orderETA'] = eta
        self._json['comments'] = comments
        self._json['visit'] = visit
        self._json['shift'] = shift
        self._json['cookAssigned'] = cook
        self._json['start'] = start_time
        self._json['end'] = end_time
        Order.id += 1
        self._json['id'] = Order.id

        print(self.json)
